src/JsonSerializationWriter.php

```diff
@@ -310,43 +310,62 @@ public function writeAnyValue(?string $key, $value): void{
         $type = get_debug_type($value);
         switch ($type) {
             case 'float':
-                $this->writeFloatValue($key, $value);
+                if (is_float($value))
```

Comment:
If we can only be 100% sure of $value's type using `is_float(), is_string() etc`, perhaps we don't need to `get_debug_type` then switch? Perhaps only if-else statements?